src/Angor/Client/Services/EncryptionService.cs

```diff
@@ -20,5 +24,26 @@ public async Task<string> DecryptData(string encryptedData, string password)
         {
             return await _jsRuntime.InvokeAsync<string>("decryptData", encryptedData, password);
         }
+
+        public async Task<string> EncryptNostrContentAsync(string nsec, string npub, string content)
+        {
+            var secertHex = GetSharedSecretHexWithoutPrefix(nsec, npub);
+            return await _jsRuntime.InvokeAsync<string>("encryptNostr", secertHex, content);
+        }
+
+        public async Task<string> DecryptNostrContentAsync(string nsec, string npub, string encryptedContent)
+        {
+            var secertHex = GetSharedSecretHexWithoutPrefix(nsec, npub);
+            return await _jsRuntime.InvokeAsync<string>("decryptNostr", secertHex, encryptedContent);
+        }
+
+        private static string GetSharedSecretHexWithoutPrefix(string nsec, string npub)
+        {
+            var privateKey = new Key(Encoders.Hex.DecodeData(nsec));
+            var publicKey = new PubKey("02" + npub);
```

Comment:
did we confirm this is true in all cases?
Can you add the link as comment to where we discovered the "02" const prefix?

Reply:
This is the way it is done in nostr-tools, it's hard coded to 02 in there.